Update `slot_usage` for `git_url` in the `MetabolomicsAnalysis` class.

brynnz22 opened the pull request:
This PR addresses issue: https://github.com/microbiomedata/nmdc-schema/issues/1942

It updates the slot_usage in the MetabolomicsAnalysis class for the git_url slot, so it clarifies that the url must point to an exact workflow for workflows that share the same github repo.

kheal commented on src/schema/workflow_execution_activity.yaml:
```diff
@@ -226,6 +226,14 @@ classes:
+        comments: >-
+          The git_url must point to an exact workflow if multiple workflows share the
+          same github repo. For instance, to differentiate lipids from GC-MS metabolomics 
+          and LCMS metabolomics in the MetaMS repo.
```
Should be "LC-MS" metabolomics.

turbomam commented on src/schema/workflow_execution_activity.yaml:
```diff
@@ -226,6 +226,14 @@ classes:
+        description: >-
+          The url that points to the github location of the metabolomics workflow.
+        comments: >-
+          The git_url must point to an exact workflow if multiple workflows share the
```
This is good guidance, but I would prefer not to be customizing slots so much To what degree can we generalize this rule over the whole schema?

turbomam replied:
Please give some examples of exact an inexact workflows. if you can't think of a  good place to put that in the schema, just paste them into comments here.

brynnz22 replied:
@kheal  can you give an example of these exact and inexact workflows. I tried to clarify in the `comments`. @turbomam  are you saying you would rather add this comment to the global definition of `git_url`?

turbomam replied:
Are you comfortable checking for patterns in existing data to look for patterns?

Can I see some source data (that may not be loaded in MongoDB yet?) that shows where things have been done with good or insufficient exactness?

turbomam replied:
> are you saying you would rather add this comment to the global definition of `git_url`

I want us to consider that kind of thing any time we make any change to the schema, especially `slot_usage`s, `rules`, etc.

THis kind of request should always be supported by data, either in the schema, as part of the PR, or in example files.

turbomam replied:
EMSL people are the main contributors to the schema nowadays, but I am concerned that the schema is being disproportionately customized for spectromety assays and workflows. It's especially difficult because I'm not able to see the data or the code.

kheal replied:
Currently the github repo [MetaMS](https://github.com/microbiomedata/metaMS) houses the GC-MS metabolomics workflow.  In the (near) future, it will also house the LC-MS lipidomics workflow and (further in the future) the LC-MS metabolomics workflows.  So pointing to `https://github.com/microbiomedata/metaMS` is insufficient, since these are all examples of workflows used for `MetabolomicsAnalysis`.  Instead, we should point to this url to specify which workflow from this repo as used `https://github.com/microbiomedata/metaMS/blob/master/metaMS/gcmsWorkflow.py`.

brynnz22 replied:
@turbomam I added the examples Katherine gave to the comments

turbomam replied:
Thanks. I wonder if we should use an enumeration here? I can't remember if switching from a string range in the global slot definition to an enumeration in a slot usage is considered monotonic.

turbomam replied:
Can one of you please check with some non-EMSL people about their practices for populating the `git_url` slot? Maybe @scanon ?

brynnz22 replied:
I am thinking we would want to leave it as a string to make space for other git_urls that come from outside of emsl. As you mentioned, you are worried about making the schema to specific to emsl - I think make this an enumeration would create too much specificity.

turbomam commented:
I guess `https://github.com/microbiomedata/metaMS/blob/master/metaMS/gcmsWorkflow.py` or `https://raw.githubusercontent.com/microbiomedata/metaMS/master/metaMS/gcmsWorkflow.py` should be an example value in this `slot_usage`.

brynnz22 commented:
> I guess `https://github.com/microbiomedata/metaMS/blob/master/metaMS/gcmsWorkflow.py` or `https://raw.githubusercontent.com/microbiomedata/metaMS/master/metaMS/gcmsWorkflow.py` should be an example value in this `slot_usage`.

I can add the valid one as an example value. But I don't think the invalid should be an example value. That's why I added them to the comments as you suggested.

brynnz22 commented:
I added the valid example as and example value.

brynnz22 commented:
@turbomam As discussed - I added a global definition to `git_url` and tried to leave out any emsl specific comments, but still maintain that we want and _exact_ github location. I also added the example to the global def.

kheal commented on src/schema/basic_slots.yaml:
```diff
@@ -81,9 +81,11 @@ slots:
       - prov:endedAtTime
 
   git_url:
+    description: The url the points to the exact github location of a workflow.
```
Small typo in here - The url "that" points to the exact github location....

brynnz22 replied:
Thank you! I fixed it!

turbomam commented on src/schema/basic_slots.yaml:
```diff
@@ -81,9 +81,11 @@ slots:
       - prov:endedAtTime
 
   git_url:
+    description: The url that points to the exact github location of a workflow.
```
thanks. let's discuss this with other people who run workflows in the near future.
